server/shadow/shadow_client.c

```diff
@@ -981,9 +978,7 @@ void* shadow_client_thread(rdpShadowClient* client)
 				shadow_client_send_surface_update(client);
 			}
 
-			EnterSynchronizationBarrier(&(subsystem->barrier), 0);
-
-			while (WaitForSingleObject(UpdateEvent, 0) == WAIT_OBJECT_0);
+			shadow_multiclient_consume(UpdateSubscriber);
```

Comment:
shadow_multiclient_consume result is not checked. I guess there is a reason why you made it BOOL ?